lib/socks.c

```diff
@@ -527,12 +527,22 @@ CURLcode Curl_SOCKS5(const char *proxy_user,
     len = 0;
     socksreq[len++] = 1;    /* username/pw subnegotiation version */
     socksreq[len++] = (unsigned char) proxy_user_len;
-    if(proxy_user && proxy_user_len)
+    if(proxy_user && proxy_user_len) {
+      if(len + proxy_user_len >= sizeof(socksreq)) {
```

Comment:
I'm probably paranoid, but `proxy_user_len` even needs needs to be < `UCHAR_MAX` (255), not only < `sizeof(socksreq)` (600), doesn't it?

Reply:
Doh, of course. Thanks, I wasn't thinking straight. I'll make another attempt!